statsmodels/stats/multicollinearity.py

```diff
@@ -0,0 +1,460 @@
+    """class for multicollinearity measures for sequence of variables
+
+    This takes the order of columns as relevant and calculates statistics for
+    a column based on information in all previous columns.
+
+    see class MultiCollinear where all variables are treated symmetrically.
+
+    Parameters
+    ----------
+    data : array_like
+        data with observations in rows and variables in columns. This is ignored if
+        moment_matrix is not None. Data can be of smaller than full rank.
+    moment_matrix : None or ndarray
+        Optional moment or correlation matrix to be used for the analysis. If it is provided
+        then data is ignored.
+        The moment_matrix needs to be nonsingular because Cholesky decomposition is used.
+    standardize : bool, default True
+        If true and data are provided, then the data will be standardized to mean zero and
+        standard deviation equal to one, which is equivalent to using the correlation matrix.
+        TODO: This might be replaced by separate demean option. Variance inflation factor
+        can be calculated for data that is scaled but not demeaned.
+        See Notes in class MultiCollinear
+
+    """
+
+    def __init__(self, data, moment_matrix=None, standardize=True):
+
+        if hasattr(data, 'columns'):
+            self.columns = data.columns
+        else:
+            self.columns = None
+
+        if data is not None:
+            x = np.asarray(data)
+
+        if moment_matrix is not None:
+            xm = np.asarray(moment_matrix)
+            if standardize:
+                # check if we have correlation matrix,
+                # we cannot demean but we can scale
+                # (We could demean if const is included in uncentred moment matrix)
+                if (np.diag(xm) != 1).any():
+                    xstd = np.sqrt(np.diag(xm))
+                    xm = xm / np.outer(xstd, xstd)
+
+            triu = np.linalg.cholesky(xm).T
+
+        else:
+            if standardize:
+                x = (x - x.mean(0)) / x.std(0)
+                if np.any(np.ptp(x, axis=0) == 0):
+                    # TODO: change this? detect from xcorr, drop constant ?
+                    raise ValueError('If standardize is true, then data should ' +
+                                     'not include a constant')
+            triu = np.linalg.qr(x, mode='r')
+        # Note: we only need elementwise squares, signs in qr are irrelevant
+        self.k_vars = triu.shape[1]
+        self.triu2 = triu**2
+
+    @cache_readonly
+    def vif(self):
+        """Variance inflation factor.
+        """
+        return self.tss / self.rss
+
+    @property
+    def tss(self):
+        """Total sum of squares of squares in sequence of regression"""
+        return self.triu2.sum(0)
+
+    @property
+    def rss(self):
+        """Residual sum of squares of squares in sequence of regression"""
+        return np.diag(self.triu2)
```

Comment:
add also `ess` 

(messy names. ssr, sst, sse would be better in regression results and here)